frontends/concrete-python/tests/compilation/test_circuit.py

```diff
@@ -740,3 +740,29 @@ def f(x, y):
     circuit.keys = fhe.Keys.deserialize(keys2)
     result = circuit.decrypt(output)
     assert np.array_equal(result, (sample_x + sample_y) ** 2)
+
+
+def test_simulate_encrypt_run_decrypt(helpers):
+    """
+    Test `simulate_encrypt_run_decrypt` configuration option.
+    """
+
+    def f(x, y):
+        return x + y
+
+    inputset = fhe.inputset(fhe.uint3, fhe.uint3)
+    configuration = helpers.configuration().fork(
+        fhe_execution=False,
+        fhe_simulation=True,
+        simulate_encrypt_run_decrypt=True,
+    )
+
+    compiler = fhe.Compiler(f, {"x": "encrypted", "y": "encrypted"})
+    circuit = compiler.compile(inputset, configuration)
+
+    sample_x, sample_y = 3, 4
+    encrypted_x, encrypted_y = circuit.encrypt(sample_x, sample_y)
+    encrypted_result = circuit.run(encrypted_x, encrypted_y)
+    result = circuit.decrypt(encrypted_result)
```

Comment:
should we check that really we make things in simulation? ideally yes
I don't know how: maybe by taking a circuit which would take hours in FHE and is fast in simulation, and assert that execution time is small

Reply:
I don't think we can find a good balance of that. I have another idea though.
```python
    assert hasattr(circuit, "simulator")
    assert not hasattr(circuit, "server")

    assert isinstance(encrypted_x, int)
    assert isinstance(encrypted_y, int)

    assert isinstance(encrypted_result, int)
```
This check that the encrypted compilation never happened.

Reply:
yes!